### DISCUSSION TITLE
Better way to test for folder's existence

### DISCUSSION THREAD

**Discussion by NotAnEvilWaffle** [SilentVoid13/Templater#567] [Help] [answered]
Is there a better way to test if a folder exists besides just trying to create a folder and catching the error that would be sent about the folder already existing?
```js
        try{
            await app.vault.createFolder("TestFolder");
        }
        catch(error){
            console.log("Folder already exists")
        }
```

I've seen from past discussions that using the exists() function works too, but I have yet to get it to succeed. It would catch an actual file, but not a folder. I did try placing slashes before and after just to see if that was something important, but no luck.

![image](https://user-images.githubusercontent.com/23061090/155218161-ca3dbe0f-8157-4fd9-9041-9817dcf6af71.png)

I am 87% sure I am just being dumb, but a little help would be appreciated.

**Comment by AB1908** [comment, marked as answer]
`await app.vault.adapter.exists(newFilePath)` should do what you're looking for. Note that adapter behaves differently on mobile at the time of writing.

**Reply by NotAnEvilWaffle** [reply]
Forgot about this. Thank you it worked like a charm.
